targets/TARGET_ARM_FM/TARGET_FVP_MPS2/us_ticker.c

```diff
@@ -16,77 +16,96 @@
 #include <stddef.h>
 #include "us_ticker_api.h"
 #include "PeripheralNames.h"
-
 #define US_TICKER_TIMER1      CMSDK_DUALTIMER1
 #define US_TICKER_TIMER2      CMSDK_DUALTIMER2
 #define US_TICKER_TIMER_IRQn  DUALTIMER_IRQn
 
-int us_ticker_inited = 0;
+/** mbed OS HAL API defined us_ticker as an increment ticker
+ *  MPS2 platform provided in SSE-200 are decrement tickers
+ *  with interrupt fired counter reaches 0.
+ *
+ *  So 2 Timers are used to construct mbed OS HAL ticker.
+ *  TIMER1 is for counting, and returns inverted binary when read from it
+ *  TIMER2 is for generate interrupts
+ */
+static int us_ticker_inited = 0;
 
 void us_ticker_init(void)
 {
     if (us_ticker_inited) {
+        us_ticker_disable_interrupt();
         return;
     }
-    us_ticker_inited = 1;
 
-    US_TICKER_TIMER1->TimerControl = 0x0; // disable timer
-    US_TICKER_TIMER2->TimerControl = 0x00; // disable timer
-    US_TICKER_TIMER1->TimerLoad = 0xFFFFFFFF;
-    US_TICKER_TIMER2->TimerLoad = 0xFFFFFFFF;
+    US_TICKER_TIMER1->TimerControl  = 0x0ul; // disable TIMER1 and reset all control
+    US_TICKER_TIMER2->TimerControl  = 0x0ul; // disable TIMER2 and reset all control
+
+    US_TICKER_TIMER1->TimerLoad     = 0xFFFFFFFFul;
+    US_TICKER_TIMER2->TimerLoad     = 0xFFFFFFFFul;
+
+    US_TICKER_TIMER1->TimerControl |= CMSDK_DUALTIMER1_CTRL_SIZE_Msk; // set TIMER1 to 32 bit counter
+    US_TICKER_TIMER2->TimerControl |= CMSDK_DUALTIMER2_CTRL_SIZE_Msk; // set TIMER2 to 32 bit counter
 
-    US_TICKER_TIMER1->TimerControl = 0x62; // enable interrupt and set to 32 bit counter and set to periodic mode
-    US_TICKER_TIMER2->TimerControl = 0x42; // enable interrupt and set to 32 bit counter
+    US_TICKER_TIMER1->TimerControl |= 0x1 << CMSDK_DUALTIMER1_CTRL_PRESCALE_Pos; // set TIMER1 with 4 stages prescale
+    US_TICKER_TIMER2->TimerControl |= 0x1 << CMSDK_DUALTIMER2_CTRL_PRESCALE_Pos; // set TIMER2 with 4 stages prescale
 
-    US_TICKER_TIMER1->TimerControl |= 0x80; // enable counter
-    US_TICKER_TIMER2->TimerControl |= 0x80; // enable counter
+    US_TICKER_TIMER2->TimerControl |= CMSDK_DUALTIMER2_CTRL_MODE_Msk;     // set TIMER2 periodic mode
```

Comment:
If periodic mode means that counter after reaching 0 is reloaded and continue counting, then it looks like this mode should be used for TIMER1 which is dedicated to count elapsed time.

Reply:
As I previously mentioned in Wilfried's comment:
Based on SSE-200 [TRM](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0479d/CHDFCJFI.html)
the timer can be set in 2 main modes:
Free-running mode: which not triggers interrupts, and wraps counter form the MAX.
Periodic mode: generates interrupts when it reaches 0, and reload from TIMERLOAD value.

So this case, we use TIMER1 in the free-running mode for monitoring time eclipsing, and TIME2 in periodic mode, and reset TIMERLOAD when we set the interrupt.

maybe I should be put these as comments in the code, seem just reading from the mode name, it is a bit misleading.